toolchain/check/context.cpp

```diff
@@ -449,7 +449,11 @@ auto Context::LookupNameInCore(SemIRLoc loc, llvm::StringRef name)
   auto inst_id = LookupNameInExactScope(loc, name_id, core_package_id,
                                         name_scopes().Get(core_package_id));
   if (!inst_id.is_valid()) {
-    DiagnoseNameNotFound(loc, name_id);
+    CARBON_DIAGNOSTIC(
+        CoreNameNotFound, Error,
+        "Name `Core.{0}` implicitly referenced here, but not found.",
+        SemIR::NameId);
+    emitter_->Emit(loc, CoreNameNotFound, name_id);
```

Comment:
I'm trying to find this new diagnostic in a test file, is it untested? I understand you're not changing the control flow, but if you're changing the diagnostic maybe that should be tested?

Reply:
Added tests in #4276.